server/src/domain/user/response-dto/user-response.dto.ts

```diff
@@ -1,10 +1,28 @@
-import { UserEntity } from '@app/infra/entities';
+import { UserAvatarColor, UserEntity } from '@app/infra/entities';
+import { Optional } from '@nestjs/common';
+import { ApiProperty } from '@nestjs/swagger';
+import { IsEnum } from 'class-validator';
+
+export const getRandomAvatarColor = (user: UserEntity): UserAvatarColor => {
+  const values = Object.values(UserAvatarColor);
+  const randomIndex = Math.floor(
+    user.email
+      .split('')
+      .map((letter) => letter.charCodeAt(0))
+      .reduce((a, b) => a + b, 0) % values.length,
+  );
+  return values[randomIndex] as UserAvatarColor;
+};
 
 export class UserDto {
   id!: string;
   name!: string;
   email!: string;
   profileImagePath!: string;
+  @IsEnum(UserAvatarColor)
+  @Optional()
+  @ApiProperty({ enumName: 'UserAvatarColor', enum: UserAvatarColor })
+  avatarColor!: UserAvatarColor | null;
```

Comment:
Under what condition will this be null? Line 46 says "if null pick a value". Is this just to avoid some mobile compat issue?

Reply:
Else there's a type error with `const providedPassword = await ask(admin);` https://github.com/immich-app/immich/blob/ac7e8bcdf469bfbe9810c8ccfc39ebb1e6b37b45/server/src/domain/user/user.service.ts#L108-L120

Reply:
That would be because it is returning an user entity not a user response. Can you change it to `await ask(mapUser(admin))` instead? And then remove the null type on the response dto.

Reply:
Damn, I didn't think about this trick. Thanks !